src/Api/HttpApi.php

```diff
@@ -29,7 +30,7 @@ abstract class HttpApi
     /**
      * The HTTP client.
      *
-     * @var ClientInterface
+     * @var ClientInterface|PluginClient
```

Comment:
What is the vendor of this Client?

```
use Http\Client\Common\PluginClient;
```

The HTTP-Plug provides the `Psr\Http\...` namespace, and I don't think that is a good idea to unbind the HttPlug package or keep the constructor open to other clients that don't follow the PSR implementation.

Remove the type hint as part of the constructor looks wrong to me, especially because PHP 7 has been introducing more and more type hints everywhere.

Reply:
It is sad that the current verison of PHP still do not support union of types yet. However, I see the next release on Nov 2020 will have it! Yay. https://stackoverflow.com/a/58826132 

Until then, I think we have to do the manual workaround I have down here in the constructor ? I can add a @FIXME or something to remind later.

Reply:
I guess that we can keep the current conditional to ensure we don't inject anything unexpected. Maybe the only relevant comment is to write the Exception message in a single line rather than multiline, but that's a minor point.

Reply:
Yeah. I think it was `pretty fix` doing that magic there!